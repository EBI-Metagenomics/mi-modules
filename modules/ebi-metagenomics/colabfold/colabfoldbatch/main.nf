process COLABFOLD_COLABFOLDBATCH {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/colabfold:1.5.5--pyh7cba7a3_0':
        'biocontainers/colabfold:1.5.5--pyh7cba7a3_0' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("${meta.id}"), emit: out_dir
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    colabfold_batch \\
        $args \\
        ${fasta} \\
        ${prefix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        colabfold: 1.5.2
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir -p ${prefix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        colabfold: 1.5.2
    END_VERSIONS
    """
}
