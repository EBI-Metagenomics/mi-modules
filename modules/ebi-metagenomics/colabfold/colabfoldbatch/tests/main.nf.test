nextflow_process {

    name "Test Process COLABFOLD_COLABFOLDBATCH"
    script "../main.nf"
    process "COLABFOLD_COLABFOLDBATCH"

    tag "modules"
    tag "modules_nfcore"
    tag "colabfold"
    tag "colabfold/colabfoldbatch"

    test("small fasta protein") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/input_sequence.fa", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.out_dir
                    with(process.out.out_dir) {
                        assert size() == 1
                        with(get(0).get(1)) {
                            File folder = new File(get(0).get(1))
                            File[] listOfFiles = folder.listFiles()
                            listOfFiles = listOfFiles.sort { it.name }
                            assert listOfFiles.length == 20
                        }
                    }
                },
                { assert snapshot(process.out.versions).match('versions') }
            )
        }

    }

    test("stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/input_sequence.fa", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
