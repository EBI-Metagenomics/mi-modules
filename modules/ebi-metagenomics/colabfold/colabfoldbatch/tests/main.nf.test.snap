{
    "versions": {
        "content": [
            [
                "versions.yml:md5,66e451357b0b69c85226e5ed06a2bf57"
            ]
        ],
        "timestamp": "2024-01-09T11:34:07.877141469"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,66e451357b0b69c85226e5ed06a2bf57"
                ],
                "out_dir": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,66e451357b0b69c85226e5ed06a2bf57"
                ]
            }
        ],
        "timestamp": "2024-01-09T11:34:40.39274736"
    }
}